lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -2532,6 +2532,36 @@ impl ClientDriveQueryDirectoryResponse {
         io_status: NTSTATUS,
         buffer: Option<FsInformationClass>,
     ) -> RdpResult<Self> {
+        // This match block ensures that the passed parameters are in a configuration that's
+        // explicitly supported by the length calculation (below) and the self.encode() method.
+        match io_status {
+            NTSTATUS::STATUS_SUCCESS => {
+                if buffer.is_none() {
```

Comment:
```suggestion
            NTSTATUS::STATUS_SUCCESS if buffer.is_none() => {
``` 

Up to you whether using [match guards](https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html) makes this read cleaner.